src/activities/assignments/AssignmentEntity.js

```diff
@@ -1,13 +1,16 @@
-'use strict';
-
-import { Entity } from '../../es6/Entity.js';
-
-/**
- * AssignmentEntity class representation of a d2l Assignment.
- */
-export class AssignmentEntity extends Entity {
-	name() {
-		return this._entity && this._entity.properties && this._entity.properties.name;
-	}
-}
-
+'use strict';
+
+import { Entity } from '../../es6/Entity.js';
+
+/**
+ * AssignmentEntity class representation of a d2l Assignment.
+ */
+export class AssignmentEntity extends Entity {
+	name() {
+		return this._entity && this._entity.properties && this._entity.properties.name;
+	}
+	canEditName() {
+		return this._entity && this._entity.hasActionByName('update-name');
```

Comment:
super complex implementation I know ;-), but i guess we could try and get into the habit of writing unit tests... of course then we write them for the easy stuff and never for the harder stuff...but...